### ISSUE TITLE
Issue with at-avx and indices

### ISSUE THREAD

**Issue by ranocha** [JuliaSIMD/LoopVectorization.jl#262]
I get the following error.
```julia
julia> using LoopVectorization

julia> function multiply_dimensionwise!(data_out::AbstractArray{<:Any, 3}, matrix::AbstractMatrix,
                                        data_in:: AbstractArray{<:Any, 3},
                                        tmp1=zeros(eltype(data_out), size(data_out, 1), size(matrix, 1), size(matrix, 2)))

         @avx for j in indices((tmp1, data_in), (3, 3)), i in indices((tmp1, matrix), (2, 1)), v in indices((tmp1, data_in), (1, 1))
           res = zero(eltype(tmp1))
           for ii in indices((data_in, matrix), (2, 2))
             res += matrix[i, ii] * data_in[v, ii, j]
           end
           tmp1[v, i, j] = res
         end

         @avx for j in indices((data_out, matrix), (3, 1)), i in indices((data_out, tmp1), (2, 2)), v in indices((data_out, tmp1), (1, 1))
           res = zero(eltype(data_out))
           for jj in indices((tmp1, matrix), (3, 2))
             res += matrix[j, jj] * tmp1[v, i, jj]
           end
           data_out[v, i, j] = res
         end

         return nothing
       end
multiply_dimensionwise! (generic function with 2 methods)

julia> data_out = randn(4, 4, 4); matrix = randn(4, 4); data_in = randn(4, 4, 4);

julia> multiply_dimensionwise!(data_out, matrix, data_in)
ERROR: BoundsError: attempt to access Tuple{Int64, Int64} at index [3]
Stacktrace:
 [1] getindex
   @ ./tuple.jl:29 [inlined]
 [2] check_match(pm::Matrix{Bool}, Rₙ::Tuple{Int64, Int64, Int64}, Sₙ::Tuple{Int64, Int64, Int64}, Xₙ::Core.SimpleVector, Dₙ::Tuple{Bool, Bool, Bool}, j::Int64, Rₚ::Tuple{Int64, Int64}, Sₚ::Tuple{Int64, Int64}, Xₚ::Core.SimpleVector, Dₚ::Tuple{Bool, Bool}, k::Int64)
   @ VectorizationBase ~/.julia/packages/VectorizationBase/bYx3Z/src/strided_pointers/grouped_strided_pointers.jl:99
 [3] #s144#58
   @ ~/.julia/packages/VectorizationBase/bYx3Z/src/strided_pointers/grouped_strided_pointers.jl:186 [inlined]
 [4] var"#s144#58"(P::Any, C::Any, B::Any, R::Any, X::Any, O::Any, D::Any, G::Any, ::Any, ptrs::Any, contig_axis::Any, batch_sz::Any, r::Any, x::Any, o::Any, d::Any, #unused#::Any)
   @ VectorizationBase ./none:0
 [5] (::Core.GeneratedFunctionStub)(::Any, ::Vararg{Any, N} where N)
   @ Core ./boot.jl:571
 [6] grouped_strided_pointer
   @ ~/.julia/packages/VectorizationBase/bYx3Z/src/strided_pointers/grouped_strided_pointers.jl:35 [inlined]
 [7] multiply_dimensionwise!(data_out::Array{Float64, 3}, matrix::Matrix{Float64}, data_in::Array{Float64, 3}, tmp1::Array{Float64, 3}) (repeats 2 times)
   @ Main ./REPL[3]:5
 [8] top-level scope
   @ REPL[5]:1
```
If I remove `@avx`, I don't get an error.
```julia
julia> function multiply_dimensionwise!(data_out::AbstractArray{<:Any, 3}, matrix::AbstractMatrix,
                                        data_in:: AbstractArray{<:Any, 3},
                                        tmp1=zeros(eltype(data_out), size(data_out, 1), size(matrix, 1), size(matrix, 2)))

         for j in indices((tmp1, data_in), (3, 3)), i in indices((tmp1, matrix), (2, 1)), v in indices((tmp1, data_in), (1, 1))
           res = zero(eltype(tmp1))
           for ii in indices((data_in, matrix), (2, 2))
             res += matrix[i, ii] * data_in[v, ii, j]
           end
           tmp1[v, i, j] = res
         end

         for j in indices((data_out, matrix), (3, 1)), i in indices((data_out, tmp1), (2, 2)), v in indices((data_out, tmp1), (1, 1))
           res = zero(eltype(data_out))
           for jj in indices((tmp1, matrix), (3, 2))
             res += matrix[j, jj] * tmp1[v, i, jj]
           end
           data_out[v, i, j] = res
         end

         return nothing
       end

julia> data_out = randn(4, 4, 4); matrix = randn(4, 4); data_in = randn(4, 4, 4); tmp1=zeros(eltype(data_out), size(data_out, 1), size(matrix, 1), size(matrix, 2));

julia> multiply_dimensionwise!(data_out, matrix, data_in)
```

**Comment by chriselrod** [comment]
Thanks for the issue. Fixed with VectorizationBase 0.20.3 (which is now set as the lower bound).

**Comment by ranocha** [comment]
Thanks for the fast fix!
